Packs/EWS/Integrations/EWSv2/EWSv2.py

```diff
@@ -904,18 +908,25 @@ def fetch_last_emails(account, folder_name='Inbox', since_datetime=None, exclude
         qs = qs.filter(datetime_received__gte=since_datetime)
     else:
         if not FETCH_ALL_HISTORY:
-            last_10_min = EWSDateTime.now(tz=EWSTimeZone.timezone('UTC')) - timedelta(minutes=10)
-            qs = qs.filter(datetime_received__gte=last_10_min)
+
+            tz = EWSTimeZone.timezone('UTC')
+            first_fetch_datetime = dateparser.parse(FETCH_TIME)
+            first_fetch_ews_datetime = EWSDateTime.from_datetime(tz.localize(first_fetch_datetime))
+            qs = qs.filter(datetime_received__gte=first_fetch_ews_datetime)
     qs = qs.filter().only(*map(lambda x: x.name, Message.FIELDS))
     qs = qs.filter().order_by('datetime_received')
 
-    result = qs.all()
-    try:
-        result = [item for item in result if isinstance(item, Message)]
-    except ValueError as exc:
-        future_utils.raise_from(ValueError(
-            'Got an error when pulling incidents. You might be using the wrong exchange version.'
-        ), exc)
+    result = []
+    for item in qs:
+        try:
+            if isinstance(item, Message):
+                result.append(item)
+                if len(result) >= MAX_FETCH:
```

Comment:
same here

Reply:
What will happen to the rest of the incidents?
Let say you received 60 incidents and your MAX_FETCH is 50.
What will happen to the extra 10?

Reply:
In this case, we will take the first 50, sorted by `datetime_received` and in the next time we will pull incidents starting from the last incident time we pulled in the previous fetch.